src/browser-polyfill.js

```diff
@@ -491,19 +491,24 @@ if (typeof browser === "undefined" || Object.getPrototypeOf(browser) !== Object.
       get: {minArgs: 1, maxArgs: 1},
       set: {minArgs: 1, maxArgs: 1},
     };
-    apiMetadata.privacy = {
-      network: {
-        networkPredictionEnabled: settingMetadata,
-        webRTCIPHandlingPolicy: settingMetadata,
-      },
-      services: {
-        passwordSavingEnabled: settingMetadata,
-      },
-      websites: {
-        hyperlinkAuditingEnabled: settingMetadata,
-        referrersEnabled: settingMetadata,
-      },
-    };
+
+    apiMetadata.privacy = {};
+    for (const type of ["network", "services", "websites"]) {
+      Object.defineProperty(apiMetadata.privacy, type, {
+        get: () => {
+          const origPrivacyType = extensionAPIs.privacy && extensionAPIs.privacy[type];
+          if (origPrivacyType) {
+            const privacyType = {};
+            for (const privacyName of Object.keys(origPrivacyType)) {
+              privacyType[privacyName] = settingMetadata;
+            }
+            apiMetadata.privacy[type] = privacyType;
+            return privacyType;
+          }
+        },
+        set: (value) => value,
```

Comment:
This is only used internally; no setter necessary.

Reply:
When I don't use setters, [getters only error](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Getter_only) is thrown when [caching the value](https://github.com/mozilla/webextension-polyfill/pull/205/files#diff-a8dc0200482fd536ca439ed8bcf763c2R505) as the `strict mode` is used.

Not sure why this is not being caught by tests.

**Steps to reproduce:**
1. `npm run build`.
2. Load `dist/browser-polyfill.js` into a chrome extension.
3. execute `browser.privacy.services.passwordSavingEnabled.get({});`

**Observer behavior:**
```
Uncaught TypeError: Cannot set property services of #<Object> which has only a getter
```

**Expected:**
No error is thrown.

Reply:
This is because you're triggering the setter (at line 505).

There are two ways to replace a getter property:

1. `delete` the original property and assign the value.
2. Use Object.defineProperty` to replace the property descriptor with one that only has a value.

In a unit test, you should count the number of accesses to `chrome.privacy.network` when you dereference the `chrome.privacy.network` object.

After looking at the implementation, it is apparent that the metadata is dereferenced only once and cached afterwards. I guess that you don't even have to replace the getter, so the code can be simplified.

implementation where metadata is dereferenced once:

https://github.com/mozilla/webextension-polyfill/blob/5f7a8458c9b249052816f520e4f8b13bf6337000/src/browser-polyfill.js#L256-L261

Reply:
> In a unit test, you should count the number of accesses to chrome.privacy.network when you dereference the chrome.privacy.network object.

I don't understand this comment, you want me to update the tests in a way?

> After looking at the implementation, it is apparent that the metadata is dereferenced only once and cached afterwards. I guess that you don't even have to replace the getter, so the code can be simplified

Sounds good, then I guess I can just remove the line where I overwrite the getter and that's it?

Reply:
> After looking at the implementation, it is apparent that the metadata is dereferenced only once and cached afterwards. I guess that you don't even have to replace the getter, so the code can be simplified

Done, removed the cache. Hope I understood your comment correctly.

> In a unit test, you should count the number of accesses to chrome.privacy.network when you dereference the chrome.privacy.network object.

I need a hint here, not sure how I can implement that.

Reply:
@Rob--W I have added a test, looking into existing implementation you provided, but I'm bit unsure about it.

> It would probably also be good to verify that if fakeChrome.privacy.services is undefined, that window.privacy.network.services is undefined as well

Not sure if I understood this one, are you referring to something like?
```
equal(window.browser.privacy.services, undefined);
```